src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php

```diff
@@ -0,0 +1,292 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(';');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Forbid multi-line whitespace before the closing semicolon or move the semicolon to the new line for chained calls.',
+            [
+                new CodeSample(
+                    '<?php
+function foo () {
+    return 1 + 2
+        ;
+}
+'
+                ),
+                new CodeSample(
+                    '<?php
+                        $this->method1()
+                            ->method2()
+                            ->method(3);
+                    ?>
+',
+                    ['strategy' => self::STRATEGY_NEW_LINE_FOR_CHAINED_CALLS]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(
+                    'strategy',
+                    'Forbid multi-line whitespace or move the semicolon to the new line for chained calls.'
+            ))
+                ->setAllowedValues([self::STRATEGY_NO_MULTI_LINE, self::STRATEGY_NEW_LINE_FOR_CHAINED_CALLS])
+                ->setDefault(self::STRATEGY_NO_MULTI_LINE)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::STRATEGY_NEW_LINE_FOR_CHAINED_CALLS === $this->configuration['strategy']) {
+            return $this->applyChainedCallsFix($tokens);
+        }
+
+        if (self::STRATEGY_NO_MULTI_LINE === $this->configuration['strategy']) {
+            return $this->applyNoMultiLineFix($tokens);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     */
+    protected function applyNoMultiLineFix(Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        foreach ($tokens as $index => $token) {
+            if (!$token->equals(';')) {
+                continue;
+            }
+
+            $previousIndex = $index - 1;
+            $previous = $tokens[$previousIndex];
+            if (!$previous->isWhitespace() || false === strpos($previous->getContent(), "\n")) {
+                continue;
+            }
+
+            $content = $previous->getContent();
+            if (("\n" === $content[0] || "\r" === $content[0]) && $tokens[$index - 2]->isComment()) {
+                $tokens[$previousIndex] = new Token([$previous->getId(), $lineEnding]);
+            } else {
+                $tokens->clearAt($previousIndex);
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     */
+    protected function applyChainedCallsFix(Tokens $tokens)
```

Comment:
There methods should be private as it only relates to this Fixer and is not require by parent abstract class like `applyFix()`.
It's common practise in other Fixers here.